### PR TITLE
(feat) app: add unit attack interaction (#6)

### PR DESCRIPTION
**Pull request by daralion**

### CODE REVIEW

File: Assets/Scripts/Entity/Unit/Unit.cs [hunk 1]
```diff
@@ -2,28 +2,54 @@
...
+using UnityEngine.AI;
 
 /*
     Class that defines a GameObject attached to a unit structure.
  */
 public class Unit : MonoBehaviour
 {
     public UnitEntity unitEntity;
+    public GameObject targettedEntity;
+    private float attackCooldown = 0;
```
Comment (CAMV) [inline Assets/Scripts/Entity/Unit/Unit.cs]: Maybe AttackSpeed instead of attackCoolDown for clarity

File: Assets/Scripts/Entity/Unit/Unit.cs [hunk 2]
```diff
@@ -2,28 +2,54 @@
...
     void Start()
     {
+        attackCooldown = unitEntity.attackTime;
     }
 
     // Update is called once per frame
-    void Update()
+    void FixedUpdate()
     {
+        reduceAttackTimeTick();
```
Comment (CAMV) [inline Assets/Scripts/Entity/Unit/Unit.cs]: - The attack behavior could be put in its own function.
- Also, other types of entities have the attack behavior. Delegation to this behavior or inheritance from another class could be beneficial

File: Assets/Scripts/Entity/Unit/Unit.cs [hunk 3]
```diff
@@ -37,5 +63,43 @@ public void onClick()
...
+
+    /* Performs an attack against the selected entity */
+    public void performAttack(Unit target)
+    {
+        bool isTargetAlive = target.takeDamage(unitEntity.damage);
+        attackCooldown = unitEntity.attackTime;
 
+        if (!isTargetAlive)
+        {
+            targettedEntity = null;
```
Comment (CAMV) [inline Assets/Scripts/Entity/Unit/Unit.cs]: Maybe delegate this function to the game manager, or some manager that keeps track of the objects in the scene.

File: Assets/Scripts/Entity/Unit/Unit.cs (hunk 1 above)
Comment (CAMV) [inline Assets/Scripts/Entity/Unit/Unit.cs]: Maybe AttackSpeed instead of attackCoolDown for clarity

File: Assets/Scripts/Entity/Unit/Unit.cs [hunk 4]
```diff
@@ -37,5 +63,43 @@ public void onClick()
...
+            isAttacking = false;
+        }
+    }
+
+    public bool takeDamage(int damage)
```
Comment (CAMV) [inline Assets/Scripts/Entity/Unit/Unit.cs:88]: Same as the behavior attack, other entities share this behavior.

**Comment by daralion** [comment]
issues solved. Took a component based approach to handle attack and movement, and moved unit death handling to the GameManager

**Pull request merged**